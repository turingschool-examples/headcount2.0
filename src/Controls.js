import React from 'react';

export const Controls = ({onChange}) => {

    return(
      <section className='controls-container'>
        <input className='input-field'
                type='text'
                placeholder='Search by district'
                onChange={(event) => onChange(event.target.value)}
          />
      </section>
    )
}
