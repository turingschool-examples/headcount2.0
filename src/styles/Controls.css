.search-wrap {
  text-align: center;
}

.search-wrap input{
  font-size: 20px;
}

