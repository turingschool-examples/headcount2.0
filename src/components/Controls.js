import React, { Component } from 'react';
import '../styles/Controls.css'

class Controls extends Component {
  constructor(props){
    super(props)
    this.state = {
      district: ''
    }
  }

  handleInput = (e) => {
    const { value } = e.target;
    this.setState({
      district: value
    })
    this.props.handleSearch(this.state.district)
  }

  render() {
    return (
      <header className='search-wrap'>
        <h1> HeadCount 2.0 </h1>
        <input onChange={this.handleInput} value={this.state.district} type='text' placeholder='Enter District Name' /> 
        
      </header>
      )
  }

}

export default Controls