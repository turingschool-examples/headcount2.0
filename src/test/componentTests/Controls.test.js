import React from 'react';
import ReactDOM from 'react-dom';
import Controls from '../../components/Controls';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<Controls/>)
  expect(renderedComponent).toMatchSnapshot()
})