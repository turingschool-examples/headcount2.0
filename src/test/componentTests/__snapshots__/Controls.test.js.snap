// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           HeadCount 2.0 
</h1>,
        <input
          placeholder="Enter District Name"
          type="text"
/>,
        <button>
          Search
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " HeadCount 2.0 ",
        },
        "ref": null,
        "rendered": " HeadCount 2.0 ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "placeholder": "Enter District Name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             HeadCount 2.0 
</h1>,
          <input
            placeholder="Enter District Name"
            type="text"
/>,
          <button>
            Search
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " HeadCount 2.0 ",
          },
          "ref": null,
          "rendered": " HeadCount 2.0 ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "placeholder": "Enter District Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
